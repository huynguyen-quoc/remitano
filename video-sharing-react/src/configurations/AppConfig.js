const AppConfig = {
  routePrefix: "/dashboard",
  dateFormat: "DD/MM/YYYY",
  dateTimeFormat: "DD/MM/YYYY HH:mm:ss"
};

export default AppConfig;
