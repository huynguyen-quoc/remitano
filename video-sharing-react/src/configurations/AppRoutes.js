import { lazy } from "react";

/* istanbul ignore next */
const AppRoutes = [
  {
    path: "/404",
    component: lazy(
      /* istanbul ignore next */ () => import("containers/screens/NotFound")
    )
  }
];

export default AppRoutes;
