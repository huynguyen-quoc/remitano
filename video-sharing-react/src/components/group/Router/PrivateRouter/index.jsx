import React, { Suspense } from "react";
import { ConnectedRouter } from "connected-react-router";
import { Switch } from "react-router";
import { Route, Redirect } from "react-router-dom";
import PropTypes from "prop-types";
import routerHistory from "modules/routerHistory";
import { checkPermissions } from "utils/helpers";
import { AppConfig } from "configurations";

export const LoadingMessage = () => "Loading...";

export function PrivateRouter(props) {
  const { userPermissions, routes, NotFoundPage } = props;
  return (
    <ConnectedRouter history={routerHistory}>
      <Suspense fallback={<LoadingMessage />}>
        <Switch>
          {routes.map(route => {
            const { path, exact, permission, ...rest } = route;
            const { values, type } = permission || { values: [], type: "some" };
            const permited = checkPermissions(
              userPermissions,
              values || [],
              type || "some"
            );
            return permited ? (
              <Route
                exact={exact}
                key={path}
                path={`${AppConfig.routePrefix}${path}`}
                {...rest}
              />
            ) : (
              <Redirect
                key={path}
                path={`${AppConfig.routePrefix}${path}`}
                {...rest}
                to={{
                  pathname: `${AppConfig.routePrefix}/404`
                }}
              />
            );
          })}
          <Route component={NotFoundPage} />
        </Switch>
      </Suspense>
    </ConnectedRouter>
  );
}

PrivateRouter.propTypes = {
  NotFoundPage: PropTypes.any.isRequired,
  routes: PropTypes.array.isRequired,
  userPermissions: PropTypes.array.isRequired
};

export default PrivateRouter;
