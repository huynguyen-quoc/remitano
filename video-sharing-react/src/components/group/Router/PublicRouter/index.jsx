import React from "react";
import PropTypes from "prop-types";
import { ConnectedRouter } from "connected-react-router";
import { Switch } from "react-router";
import { AppConfig } from "configurations";
import RoutePublic from "./RoutePublic";
import RoutePrivate from "./RoutePrivate";

function PublicRouter(props) {
  const {
    isAuthenticated,
    DashBoardPage,
    NotFoundPage,
    UnauthorizedPage
  } = props;
  return (
    <ConnectedRouter history={routerHistory}>
      <Switch>
        <RoutePrivate
          path={AppConfig.routePrefix}
          isAuthenticated={isAuthenticated}
          component={DashBoardPage}
        />
        <RoutePublic
          isAuthenticated={isAuthenticated}
          component={NotFoundPage}
        />
      </Switch>
    </ConnectedRouter>
  );
}

PublicRouter.propTypes = {
  DashBoardPage: PropTypes.any.isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
  UnauthorizedPage: PropTypes.any.isRequired,
  NotFoundPage: PropTypes.any.isRequired
};

export default PublicRouter;
