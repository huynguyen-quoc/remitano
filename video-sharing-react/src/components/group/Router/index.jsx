import { AppConfig } from "configurations";
import PropTypes from "prop-types";
import React from "react";
import { Route } from "react-router-dom";
import RoutePrivate from "./RoutePrivate";
import RoutePublic from "./RoutePublic";
import { Switch } from "react-router";

function PublicRouter(props) {
  const { isAuthenticated, notFoundComponent, routes } = props;
  return (
    <Switch>
      {routes.map(route => {
        const { path, exact, component, ...rest } = route;

        return (
          <RoutePrivate
            key={`${AppConfig.routePrefix}${path}`}
            path={`${AppConfig.routePrefix}${path}`}
            to={`${AppConfig.routePrefix}/404`}
            isAuthenticated={isAuthenticated}
            component={component}
            {...rest}
          />
        );
      })}
      <Route path="/"></Route>
      <RoutePublic
        isAuthenticated={isAuthenticated}
        component={notFoundComponent}
      />
    </Switch>
  );
}

PublicRouter.propTypes = {
  isAuthenticated: PropTypes.bool.isRequired,
  routes: PropTypes.array.isRequired,
  notFoundComponent: PropTypes.any.isRequired
};

export default PublicRouter;
