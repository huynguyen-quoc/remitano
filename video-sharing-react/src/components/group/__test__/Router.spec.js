import { MemoryRouter } from "react-router-dom";
import NotFoundScreen from "containers/screens/NotFound";
import React from "react";
import Router from "components/group/Router";
import { renderToString } from "react-dom/server";
import { AppRoutes as routes } from "configurations";

describe("Router", () => {
  it("should redirect for unauthenticated access", () => {
    const render = renderToString(
      <MemoryRouter initialEntries={["/private"]}>
        <Router
          routes={routes}
          notFoundComponent={NotFoundScreen}
          isAuthenticated={false}
        />
      </MemoryRouter>
    );

    expect(render).toMatchSnapshot();
  });

  it("should redirect for authenticated access", () => {
    const render = renderToString(
      <MemoryRouter initialEntries={["/private"]}>
        <Router
          routes={routes}
          notFoundComponent={NotFoundScreen}
          isAuthenticated={true}
        />
      </MemoryRouter>
    );

    expect(render).toMatchSnapshot();
  });
});
