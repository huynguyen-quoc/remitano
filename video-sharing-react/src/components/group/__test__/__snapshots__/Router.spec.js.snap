// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router should redirect for authenticated access 1`] = `ShallowWrapper {}`;

exports[`Router should redirect for unauthenticated access 1`] = `ShallowWrapper {}`;
